x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/security/authz_api_keys/20_field_level_security.yml

```diff
@@ -0,0 +1,113 @@
+---
+setup:
+  - skip:
+      features: headers
+
+  - do:
+      cluster.health:
+        wait_for_status: yellow
+
+  - do:
+      indices.create:
+        index: index_fls1
+        body:
+          mappings:
+            properties:
+              name:
+                type: "keyword"
+              secret:
+                type: "keyword"
+
+  - do:
+      indices.create:
+        index: index_fls2
+        body:
+          mappings:
+            properties:
+              name:
+                type: "keyword"
+              object:
+                type: "object"
+                properties:
+                  secret:
+                    type: keyword
+                  public:
+                    type: keyword
+---
+"Retrieve documents with FLS":
+
+  - do:
+      bulk:
+        refresh: true
+        body:
+          - '{"index": {"_index": "index_fls1", "_id": "1a" }}'
+          - '{"name": "i1_doc", "secret":"squirrel"}'
+          - '{"index": {"_index": "index_fls2", "_id": "2a"}}'
+          - '{"name": "i2_doc", "object":{ "secret":"mission", "public":"interest" }}'
+  - match: { errors: false }
+
+  - do:
+      security.create_api_key:
+        body:
+          name: "test-fls"
+          expiration: "1d"
+          role_descriptors:
+            index_access:
+              indices:
+                - names: [ "index_fls1" ]
+                  privileges: [ "read" ]
+                  field_security:
+                    grant: [ "name" ]
+                - names: [ "index_fls2" ]
+                  privileges: [ "read", "monitor" ]
+                  field_security:
+                    grant: [ "*" ]
+                    except: [ "object.secret" ]
+  - match: { name: "test-fls" }
+  - is_true: id
+  - set:
+      id: api_key_id
+      encoded: credentials
+
+  # With superuser...
+  - do:
+      search:
+        index: index_fls1
+  - match: { hits.total.value: 1 }
+  - match: { hits.total.relation: "eq" }
+  - match: { hits.hits.0._source.name: "i1_doc" }
+  - is_true: "hits.hits.0._source.secret"
+  - match: { hits.hits.0._source.secret: "squirrel" }
+
+  # With FLS API Key
+  - do:
+      headers:
+        Authorization: "ApiKey ${credentials}"
+      search:
+        index: index_fls1
+  - match: { hits.total.value: 1 }
+  - match: { hits.total.relation: "eq" }
+  - match: { hits.hits.0._source.name: "i1_doc" }
+  - is_false: "hits.hits.0._source.secret"
+
+
+  # With superuser...
+  - do:
+      get:
```

Comment:
nit: the above is a `search` against fls1 and has a flat structure, this is a `get` against fls2 and has a nested structure. both search vs. get and flat vs. nested seem relevant here. should have search and get tests against both flat and nested ?